import numpy as np
from exception import CanonException